### ISSUE TITLE
A user can explore(see) any page in admin area?

### ISSUE THREAD

**Issue by kidlj** [wagtail/wagtail#3616]
Issue:

The site structure is as following: 

```python
Homepage
  |- Blog Index
  |- Event Index
  |- People
  |- Contact Page
  | - User A
       |- Test Page 1
       |- Test Page 2
```

Now I add a user called 'UserA', who belongs to roles group of 'UserA' (the same name), and group 'UserA' is set to only has 'add' & 'publish' permissions to 'User A' page.

The left nav bar in admin area works fine, 'UserA' can only see 'User A' as the root; But when clicking on the wagtail bird icon in admin page, 'UserA' can see Homepage and all its children, like 'Blog Index' and 'Event Index'. 

So it's like a user can view any pages in Wagtail admin area? Is this on purpose?

What I really expect is a user can only see(explore) pages owned by himself, or pages that his role group has access to. It's like user isolation. Can this be done? Thanks.
  
--------------------
Project: wagtail-demo

Python: 3.6.1
Django: 1.10
Wagtail: 1.10.1

**Comment by gasman** [comment]
> But when clicking on the wagtail bird icon in admin page, 'UserA' can see Homepage and all its children, like 'Blog Index' and 'Event Index'.

Where exactly do you see these? When I access the admin dashboard page by clicking the bird icon as an ordinary editor, the only place I see pages listed are the "Your most recent edits" listing and the left nav menu.

**Comment by kidlj** [comment]
@gasman after clicking the bird icon, then click '25 Pages', you should see all pages.
![watail_pages](https://user-images.githubusercontent.com/300616/27334442-e9909d58-55fb-11e7-898e-2c1bb26de64b.png)

**Comment by gasman** [comment]
@kidlj Thanks for the clarification! I guess there are two things that need to be fixed here:

* when a user visits an explorer listing page that is outside of their [explorable root page](https://github.com/wagtail/wagtail/blob/e73c97463f71717a42ed2f13d02e2ef42c071267/wagtail/wagtailadmin/navigation.py#L18), redirect to their explorable root page
* The page count shown in the dashboard should only count pages that the user has edit permission for.

**Comment by gasman** [comment]
Fixed in #4206 and #4207
